src/main/java/bio/terra/metadata/StudyTableColumn.java

```diff
@@ -2,9 +2,12 @@
 
 import bio.terra.model.ColumnModel;
 
+import java.util.UUID;
+
 public class StudyTableColumn {
     private String name;
     private String type;
+    private UUID id;
```

Comment:
I think maybe StudyTableColumn should hold a back pointer to the study table.
Relationship is holding a reference to StudyTableColumn and I think needs to use that to backtrack to the Study.

Reply:
i'm hesitant to add the circular dependency until we know when/how it will be used.

Reply:
I think there is a hole here. We may not want to fix it right now, but we will need to fix it for implementing create dataset.

In the generated model, Relationship is between (table, column) and (table, column).

In the repo database schema, Relationship is between (column) and (column) AND the column table holds a reference to its owning table. (@thathert convinced me we didn't need the table reference in the schema because the columns knew their tables.)

In the metadata, Relationship is between (column) and (column) BUT column does not reference its owning table.

We need some way for the metadata Relationship to find the table and column. Otherwise, we cannot follow the Relationship to build the dataset. My preference is to update the database schema to make the relationship hold the id of its table, and for the metadata relationship object to hold references to the from and to tables as well as the columns. (That is largely an aesthetic argument: keeping the info the same in all of the representations.)

The other solution I see is: in the metadata have the column object hold a reference to its owner table.

Reply:
I am going to create a ticket for this. My bias towards being minimalist got the best of me and now I've been won over towards having a bias for readability. It will be easier for us in the long run to use the schema and reason about it/query it if we add the fk from the Relationship to its tables. Good to stay close to the RelationshipModel as well.

We will have introduced the possibility for the schema to be inconsistent if a relationship refers to a table and column that don't match, so we'll need to add testing around this. Wondering if there is there any way to represent this with db constraints.

Reply:
https://broadinstitute.atlassian.net/secure/RapidBoard.jspa?projectKey=DR&rapidView=517&selectedIssue=DR-125&view=planning